### PR TITLE
Preserve platform specific names in fonts.

### PR DESCRIPTION
**Pull request by ILOVEPIE**
## Description
This PR redoes the names table handling to preserve platform specific naming schemes for fonts.

## Motivation and Context
I have a project that depends on this working correctly. This closes issue #527.

## How Has This Been Tested?
All test cases pass.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I did `npm run test` and all tests passed green (including code styling checks).
- [x] I have added tests to cover my changes.
- [x] I have updated the **README** accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by ILOVEPIE** [comment]
@Connum sorry, can you re-do your review on this repository.

**Review by Connum — CHANGES_REQUESTED** [review]
As I didn't have access to the former review, I did it again quickly - hope I caught everything! :)

File: dist/opentype.js
Comment (ILOVEPIE) [reply dist/opentype.js:6072]: > Can we make this globally accessible so we can use it in different places and have a central location to extend this if required in the future?

There hasn't been any significant changes in this for many years, most new OSes nowadays just use 0, the unicode platform, really the platform is only useful for legacy OSes.

File: dist/opentype.js
Comment (ILOVEPIE) [reply dist/opentype.js:13230]: That's definitely a good idea.

File: dist/opentype.js
Comment (Connum) [reply dist/opentype.js:7339]: Just noticed I reviewed the dist file by accident... ;)

File: dist/opentype.js
Comment (Connum) [reply dist/opentype.js:13230]: woops, added that review to the dist file as well... replaced the comment for the src/font.js review

File: dist/opentype.js
Comment (Connum) [reply dist/opentype.js:6072]: Fair enough! :)

File: src/tables/sfnt.js
Comment (ILOVEPIE) [reply src/tables/sfnt.js:270]: There hasn't been any significant changes in this for many years, most new OSes nowadays just use 0, the unicode platform, really the platform is only useful for legacy OSes.

File: src/tables/name.js
Comment (ILOVEPIE) [reply src/tables/name.js:626]: There hasn't been any significant changes in this for many years, most new OSes nowadays just use 0, the unicode platform, really the platform is only useful for legacy OSes.

File: src/features/featureQuery.js
Comment (ILOVEPIE) [reply src/features/featureQuery.js:360]: This is because the Macintosh name will always exist whereas the windows one is specific to windows only, i can change it to ``(this.font.names.unicode || this.font.names.macintosh|| this.font.names.windows).fullName.en`` if you think that's better though.

Comment (ILOVEPIE) [reply src/features/featureQuery.js:360]: or actually... hmm... i could do unicode, windows, macintosh.

File: src/font.js
Comment (ILOVEPIE) [reply src/font.js:62]: Yeah, i'm going to push a change that fixes this.

File: src/tables/sfnt.js
Comment (ILOVEPIE) [reply src/tables/sfnt.js:270]: this was fixed.

**Comment by ILOVEPIE** [comment]
@Connum when you get the time, I've updated it to the latest master and fixed those issues you mentioned, needs another review.

**Comment by Connum** [comment]
I noticed that the changes break the font inspector when running `npm run start`. I made a PR in your branch to fix this.

File: src/font.js
Comment (ILOVEPIE) [reply src/font.js:62]: names isn't defined as a public API according to the typescript bindings... but I think we probably should consider it public.

Comment (Connum) [reply src/font.js:62]: The font inspector relied on it for example, and so might other software. On the other hand, there have been so many new features added and others are still in the pipeline as open PRs... Maybe with the change in maintainers it wouldn't hurt to release a new major version? But I'm not really sure...

Comment (ILOVEPIE) [reply src/font.js:62]: @Connum I notified the package that handles typescript bindings for us that the next major version will contain the changes made in the PR. I've also updated the closure compiler types.

**Review by Connum — APPROVED** [review]
Looks good!

**Comment by ILOVEPIE** [comment]
Just need this to be signed off again, as I bumped the version to 2.0.0 because of the breaking change.

**Review by Connum — APPROVED** [review]

**Comment by yne** [comment]
@ILOVEPIE Could you share the `google-closure-compiler` command line you used ?

I can't find a way to run it without heavily modifying the opentype.js source code

**Comment by ILOVEPIE** [comment]
@yne I didn't use closure compiler on opentype.js, that externs file is a file that is used by other projects to inform their closure compiler as to what functions/types/properties opentype.js exposes.

**Pull request merged**
